src/dev/event-horizon/coerce.ts

```diff
@@ -0,0 +1,15 @@
+import type { GenericParser, InferFromParser } from '../../parsers/types.js'
+
+export function coerce<T extends GenericParser>(parser: T | undefined, input: unknown): InferFromParser<T> {
+    if (parser !== undefined) {
+        if ('is' in parser) {
+            return parser.is(input) ? (input as InferFromParser<T>) : input
+        }
+        if ('_def' in parser) {
+            const parsed = parser.safeParse(input)
+            return parsed.success ? parsed.data : input
+        }
+    }
+
+    return input as InferFromParser<T>
```

Comment:
Does this play nicely with libraries you haven't included here? like `Joi` etc

Reply:
No :), because to make it generic I need to make it async, but it is only used in the arbitraries so I am fine with this compremise for now